### PR TITLE
feat: KUBERNETES-1240 - add jws authentication

### PR DESCRIPTION
**Pull request by KimNorgaard**
Adds authentication against the inventory server to the inventory client.

Use environment variables `TLS_CRT` and `TLS_KEY` to point to respective locations in the filesystem. They default to `/etc/certificates/tls.crt` and `/etc/certificates/tls.key`.

Authentication is enabled by default. Use environment variable `AUTH_ENABLED=false` to disable authentication. If the certificate files are not found it degrades gracefully and authentication will be disabled.

I also moved around the configuration to make it like inventory-server.

### CODE REVIEW

File: collect/collect.go [hunk 1]
```diff
@@ -100,17 +209,32 @@ func (c *InventoryCollection) Collect() {
...
 	log.Info().Msg("uploading inventory")
 
-	payload, err := json.Marshal(c.Inventory)
+	var (
+		payload     []byte
+		err         error
+		contentType string = "application/json; charset=UTF-8"
+	)
+
+	payload, err = json.Marshal(c.Inventory)
```
Comment (langecode) [inline collect/collect.go:218]: Just double checked - `json.Marshal` does serialize to UTF-8 encoding to this should fit the the content type: https://pkg.go.dev/encoding/json#Marshal

File: collect/collect.go [hunk 2]
```diff
@@ -100,17 +209,32 @@ func (c *InventoryCollection) Collect() {
...
 	if err != nil {
 		return errors.Wrap(err, "marshaling inventory")
 	}
 
+	if c.AuthEnabled {
+		jwt, err := c.Signer.Sign(payload)
```
Comment (langecode) [inline collect/collect.go:224]: strictly this is the `jws` - we have no jwt the payload is what normally (in OpenID Connect at least) would have been a jwt.

**Review by langecode — APPROVED** [review]
LGTM

File: collect/collect.go (hunk 2 above)
Comment (KimNorgaard) [reply collect/collect.go]: fixed

**Pull request merged**